src/main/scala/mesosphere/marathon/state/GroupManager.scala

```diff
@@ -157,7 +162,10 @@ class GroupManager @Singleton @Inject() (
       from <- rootGroup()
       (toUnversioned, resolve) <- resolveStoreUrls(assignDynamicServicePorts(from, change(from)))
       to = GroupVersioningUtil.updateVersionInfoForChangedApps(version, from, toUnversioned)
-      _ = BeanValidation.requireValid(ModelValidation.checkGroup(to, "", PathId.empty, config.maxApps.get))
+      _ = {
+        // TODO AW: test
+        validateOrThrow(V2Group(to))(validator = V2Group.v2GroupWithConfigValidator(config))
```

Comment:
Originally `ModelValidation.checkGroup` looked like this:

```scala
// TODO: Re-implement this method on it's own terms
  def checkGroup(
    group: Group,
    path: String,
    parent: PathId,
    maxApps: Option[Int]): Iterable[ConstraintViolation[V2Group]] = {

    val v2Group = V2Group(group)
    val maximumApps = maxApps.filter(group.transitiveApps.size > _).map{ num =>
      Seq(violation(v2Group, "apps", "",
        s"This Marathon instance may only handle up to $num Apps! (Override with command line option --max_apps)"))
    }.getOrElse(Seq.empty[ConstraintViolation[V2Group]])
    checkGroup(v2Group, path, parent) ++ maximumApps
  }
```
Besides the maxApps check, the main validation is done with the V2Group object in a separate `checkGroup` function. I decided to have only the V2Group validator as it contains all the major validation.

Alternatively I could make a `Group` validator, cast the group to a `V2Group` in there and call its validator. Seems actually like the right thing to do. This way the caller could validate an `Group` without explicitly transforming it to the `V2Group` (the transformation will be done within the Group validator)